src/services/io/podio/JEventProcessorPODIO.cc

```diff
@@ -234,6 +249,20 @@ void JEventProcessorPODIO::Process(const std::shared_ptr<const JEvent> &event) {
         FindCollectionsToWrite(event);
     }
 
+    // Trigger all collections once to fix the collection IDs
+    // TODO: WDC: This should not be necessary, but while we await collection IDs
+    //            that are determined by hash, we have to ensure they are reproducible
+    //            even if the collections are filled in unpredictable order (or not at
+    //            all). See also below, at "TODO: NWB:".
+    for (const auto& coll_name : m_collections_to_write) {
+        try {
+            const auto* coll_ptr = event->GetCollectionBase(coll_name);
+        }
+        catch(std::exception &e) {
+            // chomp
+        }
+    }
+
```

Comment:
FYI @nathanwbrei, this fixes collection IDs in the order of `m_collections_to_write` before letting algorithms pick them in random order.